### ISSUE TITLE
help,i can't open my website tips: tls error.

### ISSUE THREAD

**Issue by wslf1995** [cloudflare/cloudflared#1254]
i create a cloudflare tunnel it can open http website. but if i  want open a https website it will report a error.
example this :
2024-05-25T05:00:39Z ERR Request failed error="Unable to reach the origin service. The service may be down or it may not be responding to traffic from cloudflared: tls: failed to verify certificate: x509: cannot validate certificate for 192.168.122.180 because it doesn't contain any IP SANs" connIndex=3 dest=https://wafwall.wslf1995.top/ event=0 ip=198.41.192.107 type=http
who can help to fix it? 
thanks very mach.

**Comment by CocoXiaoLong** [comment]
That's not an error on the cloudflared side. Basically your cloudflared client tries to connect to this IP "192.168.122.180" but your server certificate doesn't contain this IP in it's SANs ([Subject Alternative Name](https://en.wikipedia.org/wiki/Subject_Alternative_Name)). This tells your client which IPs or DNSNames the Server should have.
I guess you have a QemuVM; "192.168.122.*" is an address range Qemu uses normally. 
If that's the case you could also just tell cloudflared to ignore the tls, since it's internal anyways.
The setting for that you can find in "**Zero Trust Dashboard>Tunnels>YOU_TUNNEL>TLS**"

**TL;DR: You can learn about x509 certificates or you just don't verify the tls internally**

<img width="1068" alt="bsf" src="https://github.com/cloudflare/cloudflared/assets/51422232/bd6f7932-a620-4c2a-bc39-c01989f64a6a">

**Comment by wslf1995** [comment]
thanks for you help.you say right.after i send this issue,i find the way to fix this error. you way is tell cloudflared to ignore the tls.in the same ,i find in "Zero Trust Dashboard>Tunnels>YOUR_TUNNEL>TLS> origin server name add my cert’s name just like xxxx.com/xxxx.top also can fix this problem.

actually, i have a QemuVM,i set ip 10.21.100.0/24 in before.it can work normal.but,lately i change it 192.168.00/24 ,it can’t work.i can’t understand how difference between 2 ip. can you explain it?
thanks again for all your help. @ CocoXiaoLong

**Comment by CocoXiaoLong** [comment]
A bit off topic @wslf1995, but ok.
First of all, I'm not really good with the networking internals of Qemu. 

About your question: What you have set is a [CIDR](https://en.wikipedia.org/wiki/Classless_Inter-Domain_Routing) IP Range.
10.21.100.0/24 translates to "All IPv4-Addresses between 10.21.100.0 and 10.21.100.255".
192.168.0.0/24 translates to "All IPv4-Addresses between 192.168.0.0 and 192.168.0.255".
Qemu uses that to act as a minimal [NAT](https://en.wikipedia.org/wiki/Network_address_translation) router between your VMs and your physical network. That's important because if say, you fail to setup a VM and try over and over again, your physical network would get clogged with non-used IP-Addresses.
In traditional networks 192.168.0.1 or 10.21.100.1 are most likely reserved for the gateway (or in other words your router) and "*.255" is reserved for broadcast (used by for example UDP to send datagrams to every device in the network).

**My guess would be that your virtual machine didn't pull a new IP from the Qemu NAT yet.**
But then again, Qemu isn't my strong point.

Back to CIDR, you might want bigger address ranges for bigger networks (say for example governments, universities, schools or really big enterprises; just a place with many devices). I mean, in that case 254 IP-Addresses are bit too little space.
For example: 10.11.0.0/16
This translates to "All between 10.11.0.0 and 10.11.255.255" which results in 65535 unique addresses.
CIDR is just a way to write as one value (10.11.0.0/16) instead of two (10.11.0.0; 10.11.255.255).

For Qemu you could assign such an address range, but then the question becomes: why. I would envy you for a machine that could handle that much VMs. After all, you would have to assign each machine a core :), and lets not speak about ram and heat in the machine.

P.S.: You might want to close the issue if it's resolved.
